src/Microsoft.ML.Core/Utilities/NormStr.cs

```diff
@@ -116,7 +116,7 @@ public NormStr Get(string str, bool add = false)
                 return add ? AddCore(str.AsMemory(), hash) : null;
             }
 
-            public NormStr Get(ReadOnlyMemory<char> str, bool add = false)
+            public NormStr Get(ReadOnlyMemory<char> str, bool add = false, bool createNewStr = true)
```

Comment:
nit -- Naming: I can foresee confusion between the two flags, `add` and `createNewStr`. Since they could both be interpreted as "if my key is missing, insert (add/create) it".

We want to convey that the string will be reallocated and no longer hold open a pointer to the original string. Perhaps as `duplicateStr` or invert as `keepAsReference`.

```suggestion
            public NormStr Get(ReadOnlyMemory<char> str, bool add = false, bool duplicateStr = true)
```